src/main/java/com/sap/sgs/phosphor/fosstars/model/score/oss/OwaspDependencyScanScore.java

```diff
@@ -0,0 +1,128 @@
+package com.sap.sgs.phosphor.fosstars.model.score.oss;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD;
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.OWASP_DEPENDENCY_CHECK_USAGE;
+import static com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage.MANDATORY;
+import static com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage.NOT_USED;
+import static com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage.OPTIONAL;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures;
+import com.sap.sgs.phosphor.fosstars.model.qa.ScoreVerification;
+import com.sap.sgs.phosphor.fosstars.model.qa.TestVectors;
+import com.sap.sgs.phosphor.fosstars.model.score.FeatureBasedScore;
+import com.sap.sgs.phosphor.fosstars.model.value.CVSS;
+import com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckCvssThresholdValue;
+import com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage;
+import com.sap.sgs.phosphor.fosstars.model.value.ScoreValue;
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.commons.lang3.Range;
+
+/**
+ * The scores assesses how well an open-source project uses OWASP Dependency Check to scan
+ * dependencies for known vulnerabilities. It is based on the following features:
+ * <ul>
+ *   <li>{@link OssFeatures#OWASP_DEPENDENCY_CHECK_USAGE}</li>
+ *   <li>{@link OssFeatures#OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD}</li>
+ * </ul>
+ */
+public class OwaspDependencyScanScore extends FeatureBasedScore {
+
+  /**
+   * The basic step score that is alloted in each step. 
+   */
+  private static final double STEP_SCORE = 3.0;
+
+  /**
+   * Initializes a new {@link OwaspDependencyScanScore}.
+   */
+  OwaspDependencyScanScore() {
+    super("How a project uses OWASP Dependency Check to scan dependencies for vulnerabilities",
+        OWASP_DEPENDENCY_CHECK_USAGE,
+        OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD);
+  }
+
+  @Override
+  public ScoreValue calculate(Value... values) {
+    Value<OwaspDependencyCheckUsage> usageValue =
+        find(OWASP_DEPENDENCY_CHECK_USAGE, values);
+    Value<Double> cvssScore =
+        find(OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD, values);
```

Comment:
Should it simplify a bit the code below?

```suggestion
    OwaspDependencyCheckUsageValue usageValue =
        find(OWASP_DEPENDENCY_CHECK_USAGE, values);
    OwaspDependencyCheckCvssThresholdValue cvssScore =
        find(OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD, values);
```

Reply:
I tried this way. But IDE suggests that we have to use casts to achieve this. Hence, felt this would be simpler.

Reply:
I appreciate IDE's suggestions. If we update the type of `OWASP_DEPENDENCY_CHECK_USAGE` to `OwaspDependencyCheckUsage`, can it make it happy?

Reply:
Furthermore, if `usageValue`is not `OwaspDependencyCheckUsageValue`, doesn't it look like that something is wrong?

Reply:
https://github.com/SAP/fosstars-rating-core/issues/278